Update sdk for GA release, version 1.1.0

juancamilor opened the pull request:
- Includes support for message metadata + new overloads for typing notification, send messages and read receipts
- Remove custom code in test to add MS-CV header. Added in core 4 months ago by default.
- API new specification PR :
https://github.com/Azure/azure-rest-api-specs/pull/15756

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

juancamilor committed:
- Update sdk for GA release, version 1.1.0 which includes message metadata + new overloads for typing notification, send messages and read receipts

- Remove custom code in test to add MS-CV header. Added in core 4 months ago by default.

jsquire commented on sdk/communication/Azure.Communication.Chat/README.md:
````diff
@@ -11,7 +11,7 @@ This package contains a C# SDK for Azure Communication Services for chat.
 
 ```PowerShell
-dotnet add package Azure.Communication.Chat --version 1.0.1
+dotnet add package Azure.Communication.Chat --version 1.1.0
````
```suggestion
dotnet add package Azure.Communication.Chat 
```

We generally recommend this form, which will use the latest GA.   That avoids the need to remember to increment the README for each release.

jsquire commented on sdk/communication/Azure.Communication.Chat/CHANGELOG.md:
```diff
@@ -1,14 +1,7 @@
 # Release History
 
-## 1.1.0-beta.2 (Unreleased)
+## 1.1.0 (Unreleased)
```
Wouldn't we want to leave an indication of what changed, even if that is just noting that its using the latest service version?

jsquire commented on sdk/communication/Azure.Communication.Chat/src/ChatClientOptions.cs:
```diff
@@ -25,7 +25,7 @@ public ChatClientOptions(ServiceVersion version = LatestVersion)
             ApiVersion = version switch
             {
-                ServiceVersion.V2021_04_05_preview6 => "2021-04-05-preview6",
+                ServiceVersion.V2021_09_07 => "2021-09-07",
```
We normally allow for older versions to be specified.   Is there a reason that we're disallowing that here and, if so, have we confirmed that it won't be viewed as a breaking change?   @tg-msft,  do you have feedback here?

If we do end up validating against older versions, we probably want to mark the older enum members with `[EditorBrowsable(EditorBrowsableState.Never)]`

juancamilor replied:
I was confused on how this should work. Got clarification in the .NET SDK channel. Thanks for pointing this out!

tg-msft replied:
We do drop preview versions when we go to GA though (because a GA client library should never talk to a preview service version which is still subject to change).  But otherwise, yes, it's important that our libraries support multiple service versions.

juancamilor committed:
Addressed comments:

- Bring back support for previous version in options
- Updated readme and changelog
- Remove preview6 support

jsquire reviewed (approved):
LGTM
